appveyor-windows-build-dotnet.yaml

```diff
@@ -0,0 +1,72 @@
+version: 1.0.{build}
+image: Windows Server 2019
+build: off
+
+# Change the clone folder to somewhere in "D:\" because this is shared by default with Docker. We need this to mount folders.
+clone_folder: D:\source
+
+environment:
+  AWS_DEFAULT_REGION: us-east-1
+  SAM_CLI_DEV: 1
+  APPVEYOR_CI_OVERRIDE: 1
+
+  # In Windows, tempdir is usually in C:\. But in AppVeyor only D:\ is shared in Docker. 
+  # Therefore change TEMPDIR it to D: for invoke tests to work. Python uses $TMPDIR envvar
+  # to find root of tempdir
+  TMPDIR: D:\tmp
+  TEMP: D:\tmp
+  TMP: D:\tmp
+
+  # MSI Installers only use Py3.6.6. It is sufficient to test with this version here.
+  PYTHON_HOME: "C:\\Python36-x64"
+  PYTHON_SCRIPTS: "C:\\Python36-x64\\Scripts"
+  PYTHON_EXE: "C:\\Python36-x64\\python.exe"
+  PYTHON_VERSION: '3.6.8'
+  PYTHON_ARCH: '64'
+  HOME: 'C:\Users\appveyor'
+  HOMEDRIVE: 'C:'
+  HOMEPATH: 'C:\Users\appveyor'
+
+init:
+  # Uncomment this for RDP
+  - ps: iex ((new-object net.webclient).DownloadString('https://raw.githubusercontent.com/appveyor/ci/master/scripts/enable-rdp.ps1'))
+  - ps: gcim Win32_Processor | % { "$($_.NumberOfLogicalProcessors) logical CPUs" }
+  - ps: gcim Win32_OperatingSystem | % { "$([int]($_.TotalVisibleMemorySize/1mb)) Gb" }
+
+
+install:
+
+    # Make sure the temp directory exists for Python to use.
+    - ps: "mkdir -Force D:\\tmp"
+    - "echo %PATH%"
+    - "python --version"
+
+    # Upgrade setuptools, wheel and virtualenv
+    - "python -m pip install --upgrade setuptools wheel virtualenv"
+
+    # Create new virtual environment with chosen python version and activate it
+    - "rm -rf venv"
+    - "dir C:\\Python36-x64"
+    - "virtualenv -p %PYTHON_EXE% venv"
```

Comment:
Instead, you can add `PYTHON_HOME` to the PATH and then python will point to the correct version. Makes this a littler easier and then things like line #45 will work on the correct python version.

Reply:
Adding to path was notoriously hard in https://github.com/awslabs/aws-sam-cli/pull/1269. There were a fair few commits in there to get binaries added to the path, and they didnt work as expected, so I skipped it this time.

Reply:
Ported over those changes.